packages/flutter/lib/src/services/text_input.dart

```diff
@@ -661,14 +660,14 @@ class TextEditingValue {
     return TextEditingValue(
       text: encoded['text'] as String,
       selection: TextSelection(
-        baseOffset: encoded['selectionBase'] as int ?? -1,
-        extentOffset: encoded['selectionExtent'] as int ?? -1,
-        affinity: _toTextAffinity(encoded['selectionAffinity'] as String) ?? TextAffinity.downstream,
-        isDirectional: encoded['selectionIsDirectional'] as bool ?? false,
+        baseOffset: encoded['selectionBase'] as int? ?? -1,
+        extentOffset: encoded['selectionExtent'] as int? ?? -1,
+        affinity: _toTextAffinity(encoded['selectionAffinity'] as String),
```

Comment:
Looking at this code again, I think the idea is that if `encoded['selectionAffinity']` is null, this should default to `TextAffinity.downstream`.